src/puzzles/rush_hour/rush_hour_block.gd

```diff
@@ -0,0 +1,62 @@
+@tool
+class_name RushHourBlock
+extends TextureRect
+
+# TODO: Find a clean way to sync this from the grid scene.
+const GRID_SIZE := 80
+
+signal selected(block: RushHourBlock)
+signal released(block: RushHourBlock)
+
+enum {
+	HORIZONTAL,
+	VERTICAL,
+}
+
+@export_enum("Horizontal:0", "Vertical:1")
+var direction: int
+
+@export_range(2,3)
+var block_size: int
+
+@export var grid_position: Vector2i = Vector2i.ZERO
+
+# Called when the node enters the scene tree for the first time.
+func _ready() -> void:
+	set_block_size()
+
+
+# Called every frame. 'delta' is the elapsed time since the previous frame.
+func _process(_delta: float) -> void:
+	if Engine.is_editor_hint():
+		set_block_size()
```

Comment:
Do we want an extra clause to the if statement so we're not running this every frame?
`if Engine.is_editor_hint() and curr_block_size != prev_block_size:`

Reply:
Eh, I don't think it matters that much tbh. This only gets run in the preview.